### ISSUE TITLE
When changing a service, the systemctl reload happens after a service restart

### ISSUE THREAD

**Issue by steveniemitz** [NTTCom-MS/eyp-systemd#35]
Assume there's a puppet manifest that looks like:
```
systemd::service { 'someservice':
 ...
 execstart => 'my command line',
 notify => Service['someservice']
}

service { 'someservice':
 ...
}
```

If I change the `execstart` command, the service is notified BEFORE the `systemctl daemon-reload` is run by the systemd module.  This is bad obviously since the service is restarted with the old parameters, not the new ones.

We've fixed this in our branch by using `contain` instead of `include`, but that's a puppet 4 only feature, and I think this want to be compatible with 3 or 4?

For reference, here's our change: https://github.com/tc-dc/eyp-systemd/commit/6c9a818a127abd220baf44b31f27f0fe5c538703

**Comment by jordiprats** [comment]
Obvious workaround is to add a require:

```
systemd::service { 'someservice':
 ...
 execstart => 'my command line',
 notify => Service['someservice'],
 require => Class['::systemd']
}

service { 'someservice':
 ...
}
```

I'd like to keep puppet3 compatibility, not sure if there's any other way to fix this

Thanks for your bug report

**Comment by davidnewhall** [comment]
I do this. Seems to fix the problem you describe. The problem is not endemic to this module though, and your work-arounds should be purely in puppet as shown below. I don't think this module has any faults in this regards.
```ruby
  service { $name:
    ensure     => $ensure,
    enable     => $enable,
    hasrestart => true,
    hasstatus  => true,
    subscribe  => Systemd::Service[$name],
    require    => Exec['systemctl reload'],
  }
```

**Comment by jordiprats** [comment]
Hi,
This was fixed for puppet4 on **0.1.34**, could you please recheck if it is working for you without this workarround?

For your reference: [889fab23d3e17e80e09742ac02b44ac91ca38329](https://github.com/jordiprats/eyp-systemd/commit/889fab23d3e17e80e09742ac02b44ac91ca38329)

thanks,
Jordi
